enhancements/api-review/add-repositoryMirrors-spec.md

```diff
@@ -0,0 +1,189 @@
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Today, the ImageContentSourcePolicy object sets up mirror with `mirror-by-digest-only` property set to true. Requires images be pulled by digest only. This enhancement plans to add new spec to ImageContentSourcePolicy with configurable field `allowMirrorByTags`, so that `mirror-by-digest-only` can be configured by ImageContentSourcePolicy.
+
+## Motivation
+
+Today, the ImageContentSourcePolicy object sets up mirror configuration with `mirror-by-digest-only` property set to true, which leads to using mirrors only when images are referenced by digests. However, when working with disconnected environments, sometimes multiple ImageContentSourcePolicies are needed, some of them are used by apps/manifests that don't use digests when pulling the images.
+
+Adding the `repositoryMirrors` spec to ImageContentSourcePolicy to configure mirrors will make it possible to pull images from mirror using tags without requiring image digest reference.
+
+### Goals
+
+- Enable the user to pull images through ImageContentSourcePolicy configured mirrors by tags.
+
+### Non-Goals
+
+- This proposal does not recommend using by-tag references for OpenShift release images. Those should still be referenced by digest, regardless of whether `allowMirrorByTags` is enabled for repositories where the release images are mirrored
+
+## Proposal
+
+The new spec `repositoryMirrors` will be added to ImageContentSourcePolicy. The `allowMirrorByTags` property will allow for mirroring by-tag images through configured mirrors.
+
+```go
+// RepositoryMirrors holds cluster-wide information about how to handle mirrors in the registries config.
+// Note: this is different from the RepositoryDigestMirrors that the mirrors only 
+// work when pulling the images that are referenced by their digests.
+type RepositoryMirrors struct {
+    // source is the repository that users refer to, e.g. in image pull specifications.
+    // +required
+    Source string `json:"source"`
+    // If true, mirrors will only be used for digest pulls. Pulling images by
+    // tag can potentially yield different images, depending on which endpoint
+    // we pull from.  Forcing digest-pulls for mirrors avoids that issue.
+    // +optional
+    AllowMirrorByTags bool `json:"allowMirrorByTags"`
+    // mirrors is one or more repositories that may also contain the same images.
+    // The order of mirrors in this list is treated as the user's desired priority, while source
+    // is by default considered lower priority than all mirrors. Other cluster configuration,
```

Comment:
Specifically "source being last" only makes sense for digest, it DOES not make sense for tag in all use cases.  Tag is fundamentally dependent on hoping you get the most up to date version of a tag during upgrade scenarios.  

Both image stream import and crio can use repository digest order consistently.  However, I'm not sure without an understanding of image import tag mirroring what we do here.

Is there another enhancement that covers ImageStream import behavior with repository tag design?  We should not be introducing a V1 api without a full design covering how import will interpret this, because mirroring of tags may actually need these feels to be designed.  If there is no additional design, I'd like to see one before I withdraw my hold on the implementation.